### PR TITLE
socket process priority change now possible for haiku since beta3 rel…

### PR DESCRIPTION
**Pull request by devnexen**
…ease.

 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

### CODE REVIEW

**Review by ret2libc — APPROVED** [review]
Thanks!

**Comment by ret2libc** [comment]
@devnexen not related to this, but could you check if everything works on Haiku if we remove the #ifdef block at https://github.com/rizinorg/rizin/blob/dev/librz/include/rz_types.h#L352 ? We would like to move away from all this "if windows or osx or haiku or ..." stuff and let just meson check which functions/headers/macros are available and use them accordingly.

**Review by wargio — APPROVED** [review]

File: librz/socket/run.c [hunk 1]
```diff
@@ -1217,7 +1217,7 @@ RZ_API int rz_run_start(RzRunProfile *p) {
 #endif
 
 		if (p->_nice) {
-#if __UNIX__ && !defined(__HAIKU__)
```
Comment (yossizap) [inline librz/socket/run.c:1220]: Is it possible to define whether this functionality is available with meson's [`compiler.run`](https://mesonbuild.com/Compiler-properties.html) during compile time? Or with some version define from a system header. Seeing that this is only available in a beta build and some people may still use older versions.

Comment (devnexen) [reply librz/socket/run.c:1220]: you re kind of forced to update to beta3 via the package manager.

Comment (ret2libc) [reply librz/socket/run.c:1220]: We want to do that in the future. In meson.build you can check if nice function exists on the host target and then have a define HAVE_NICE If it is.

Comment (devnexen) [reply librz/socket/run.c:1220]: sorry I misread the line commented, yes sure doing it.

**Review by yossizap — APPROVED** [review]

**Pull request merged**